### DISCUSSION TITLE
Pixel packing op

### DISCUSSION THREAD

**Discussion by marioepsley** [cables-gl/cables_docs#704] [Feature Requests] [answered]
The idea would be to easily pack a line of rgb pixels into a square.

<img width="473" alt="Screenshot 2024-03-21 at 12 34 16" src="https://github.com/cables-gl/cables_docs/assets/36418408/4fe22c2d-de86-4896-a9a8-db19632579d0">

^this only has a few pixels in but i'm thinking more for a larger buffer

imagine packing a buffer of 64 pixel colours neatly into an 8x8 square with a trigger.

It's almost like the the montage op

**Comment by pandrr** [comment]
hi,

i rewrote the textureRectangle op on dev.cables.gl

this can now be used to draw rectangles pixel perfect way, also it can map a texture on that rectangle. you can also choose exactly the region that will be mapped on that rectangle.
with this op a pixel packing op could be patched to gether quite easy

**Reply by marioepsley** [reply]
This is really great, I've noticed a few things,
1. the pixels have to be slightly over the integer value they should be
<img width="1041" alt="Screenshot 2024-03-22 at 12 01 54" src="https://github.com/cables-gl/cables_docs/assets/36418408/118d3986-3032-4176-95e0-e460de40c13f">
<img width="997" alt="Screenshot 2024-03-22 at 12 02 06" src="https://github.com/cables-gl/cables_docs/assets/36418408/e82fc71a-2c92-48ee-931f-cfabfa05ddec">

2. I can't seem to get the texture to work
3. Just wondering what inner and roundness do

Very useful op!!

**Comment by pandrr** [comment]
- inner makes a whole, like the circle op, to easily create "frames"
- how to map texture: width/height how much of the texture do we want to display
<img width="1876" alt="Screenshot 2024-03-22 at 13 12 26" src="https://github.com/cables-gl/cables_docs/assets/98792/a7ab0d1d-54ad-45c7-9ce6-09f282831fc4">

integer values: thx, i will check this later!

**Comment by pandrr** [comment]
<img width="626" alt="Screenshot 2024-03-22 at 13 14 52" src="https://github.com/cables-gl/cables_docs/assets/98792/94fdd9fe-79d6-43d1-a9c6-a7b2531b6f59">

**Comment by pandrr** [comment]
https://dev.cables.gl/edit/vz1GK7

there you have it

**Comment by pandrr** [comment, marked as answer]
<img width="812" alt="Screenshot 2024-03-22 at 13 38 17" src="https://github.com/cables-gl/cables_docs/assets/98792/871e6669-e5b3-4a35-847c-4c6de494d2ef">

**Comment by marioepsley** [comment]
Big thanks for this
